src/Calculator/Views/GraphingCalculator/GraphingNumPad.xaml.cpp

```diff
@@ -104,3 +193,55 @@ void GraphingNumPad::SetOperatorRowVisibility()
     Row1->Visibility = rowVis;
     InvRow1->Visibility = invRowVis;
 }
+
+void GraphingNumPad::Button_Clicked(Platform::Object ^ sender, Windows::UI::Xaml::DependencyPropertyChangedEventArgs ^ /*e*/)
+{
+    auto mathRichEdit = GetActiveRichEdit();
+    auto button = dynamic_cast<CalculatorApp::Controls::CalculatorButton ^>(sender);
+    if (mathRichEdit != nullptr && sender != nullptr)
+    {
+        auto id = button->ButtonId;
+        auto output = buttonOutput.find(id);
+        mathRichEdit->InsertText(std::get<0>(output->second), std::get<1>(output->second), std::get<2>(output->second));
+    }
+}
+
+void GraphingNumPad::SubmitButton_Clicked(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    auto mathRichEdit = GetActiveRichEdit();
+    if (mathRichEdit != nullptr)
+    {
+        mathRichEdit->SubmitEquation();
+    }
+}
+
+void GraphingNumPad::ClearButton_Clicked(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    auto mathRichEdit = GetActiveRichEdit();
+    if (mathRichEdit != nullptr)
+    {
+        mathRichEdit->MathText = L"<math xmlns=\"http://www.w3.org/1998/Math/MathML\"></math>";
+        mathRichEdit->SubmitEquation();
+    }
+
+}
+
+void GraphingNumPad::BackSpaceButton_Clicked(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    auto mathRichEdit = GetActiveRichEdit();
+    if (mathRichEdit != nullptr)
+    {
+        mathRichEdit->BackSpace();
+    }
+}
+
+// To avoid focus moving when the space between buttons is clicked, handle click events that make it through the keypad.
+void CalculatorApp::GraphingNumPad::GraphingNumPad_PointerPressed(Platform::Object ^ sender, Windows::UI::Xaml::Input::PointerRoutedEventArgs ^ e)
+{
+    e->Handled = true;
+}
+
+CalculatorApp::Controls::MathRichEditBox ^ CalculatorApp::GraphingNumPad::GetActiveRichEdit()
+{
+    return dynamic_cast<CalculatorApp::Controls::MathRichEditBox ^>(FocusManager::GetFocusedElement());
```

Comment:
I think it would be good to give the first Ghost EquationTextBox focus when going to graphing mode.

Reply:
I think Rudy has a fix for this already, but it is not present in this branch.